python/examples/extract_position_data.py

```diff
@@ -104,9 +156,12 @@
     path = os.path.join(output_dir, 'position.kml')
     logger.info("Generating '%s'." % path)
     with open(path, 'w') as f:
-        f.write(KML_TEMPLATE %
-                {'coordinates': '\n'.join(['%.8f,%.8f' % (message.lla_deg[1], message.lla_deg[0])
-                                           for message in pose_data.messages
-                                           if message.solution_type != SolutionType.Invalid])})
+        f.write(KML_TEMPLATE_START)
+        for pose in pose_data.messages:
+          f.write(KML_TEMPLATE %
+                  {'timestamp': '\n'.join([str(pose.gps_time.as_utc())]),
```

Comment:
This is not the correct timestamp format for KML. `as_utc()` returns a `datetime` object, and `str()` of that returns a string like `2023-10-05 21:22:54+00:00`, but looking at https://developers.google.com/kml/documentation/time#how-to-specify-time and https://developers.google.com/kml/documentation/time#example:-showing-movement-of-a-point-along-a-path, they're expecting a timestamp like `2007-01-14T21:05:02Z`. Note the KML docs don't show fractional seconds, but https://www.w3.org/TR/xmlschema-2/#isoformats which they reference says ss.ssss is ok.

You should be able to call `isoformat()` on the `datetime` object to get a timestamp that I believe should be in the right format.

Reply:
The `isoformat()` call ends up including a `00+00` at the end, and also doesn't include the desired `Z` a the end:
`2023-10-05T21:22:53.900000+00:00`

An alternative is to go with this call, which puts it in the format that you put:`.strftime('%Y-%m-%dT%H:%M:%S.%fZ')`

Tested both ways and they both work to incorporate the time slider.

Reply:
Does it work with the +00:00 instead of Z? They effectively mean the same thing. If it works, I'd use the builtin function. If not, the strftime call is fine.